ouroboros-consensus-protocol/src/unstable-protocol-testlib/Test/Ouroboros/Consensus/Protocol/Praos/Header.hs

```diff
@@ -0,0 +1,456 @@
+                     hashToBytes, hashWith)
+import qualified Cardano.Crypto.KES as KES
+import           Cardano.Crypto.KES.Class (genKeyKES, rawDeserialiseSignKeyKES,
+                     rawSerialiseSignKeyKES)
+import           Cardano.Crypto.Seed (mkSeedFromBytes)
+import           Cardano.Crypto.VRF (deriveVerKeyVRF, hashVerKeyVRF,
+                     rawDeserialiseSignKeyVRF, rawSerialiseSignKeyVRF)
+import qualified Cardano.Crypto.VRF as VRF
+import           Cardano.Crypto.VRF.Praos (skToBatchCompat)
+import qualified Cardano.Crypto.VRF.Praos as VRF
+import           Cardano.Ledger.BaseTypes (ActiveSlotCoeff, Nonce (..),
+                     PositiveUnitInterval, ProtVer (..), Version, activeSlotVal,
+                     boundRational, mkActiveSlotCoeff, natVersion)
+import           Cardano.Ledger.Binary (MaxVersion, decCBOR,
+                     decodeFullAnnotator, serialize')
+import           Cardano.Ledger.Keys (KeyHash, KeyRole (BlockIssuer), VKey (..),
+                     hashKey, signedDSIGN)
+import           Cardano.Protocol.TPraos.BHeader (HashHeader (..),
+                     PrevHash (..), checkLeaderNatValue)
+import           Cardano.Protocol.TPraos.OCert (KESPeriod (..), OCert (..),
+                     OCertSignable (..))
+import           Cardano.Slotting.Block (BlockNo (..))
+import           Cardano.Slotting.Slot (SlotNo (..))
+import           Data.Aeson (defaultOptions, (.:), (.=))
+import qualified Data.Aeson as Json
+import           Data.Bifunctor (second)
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Base16 as Base16
+import qualified Data.ByteString.Lazy as LBS
+import           Data.Coerce (coerce)
+import           Data.Foldable (toList)
+import qualified Data.Map as Map
+import           Data.Maybe (fromJust, fromMaybe)
+import           Data.Proxy (Proxy (..))
+import           Data.Ratio ((%))
+import           Data.Text.Encoding (decodeUtf8, encodeUtf8)
+import           Data.Word (Word64)
+import           GHC.Generics (Generic)
+import           Ouroboros.Consensus.Protocol.Praos (PraosValidationErr (..))
+import           Ouroboros.Consensus.Protocol.Praos.Header (Header,
+                     HeaderBody (..), pattern Header)
+import           Ouroboros.Consensus.Protocol.Praos.VRF (InputVRF, mkInputVRF,
+                     vrfLeaderValue)
+import           Ouroboros.Consensus.Protocol.TPraos (StandardCrypto)
+import           Test.QuickCheck (Gen, arbitrary, choose, frequency, generate,
+                     getPositive, resize, sized, suchThat, vectorOf)
+
+-- * Test Vectors
+
+generateSamples :: Int -> IO Sample
+generateSamples n = generate (resize n genSample)
+
+-- FIXME: Should be defined according to some Era
+testVersion :: Version
+testVersion = natVersion @MaxVersion
+
+data Sample = Sample {sample :: ![(GeneratorContext, MutatedHeader)]}
+    deriving (Show, Eq)
```

Comment:
Why is this not a newtype?

Reply:
I suspect this is a leftover from some refactoring.